drivers/ethernet/eth_stm32_hal.c

```diff
@@ -932,3 +1127,293 @@ static struct eth_stm32_hal_dev_data eth0_data = {
+			      struct net_ptp_time *tm)
+{
+	struct ptp_context *ptp_context = dev->data;
+	struct eth_stm32_hal_dev_data *eth_dev_data = ptp_context->eth_dev_data;
+	ETH_HandleTypeDef *heth = &eth_dev_data->heth;
+	int key;
+
+	key = irq_lock();
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	heth->Instance->MACSTSUR = tm->second;
+	heth->Instance->MACSTNUR = tm->nanosecond;
+	heth->Instance->MACTSCR |= ETH_MACTSCR_TSINIT;
+	while (heth->Instance->MACTSCR & ETH_MACTSCR_TSINIT_Msk) {
+		/* spin lock */
+	}
+#else
+	heth->Instance->PTPTSHUR = tm->second;
+	heth->Instance->PTPTSLUR = tm->nanosecond;
+	heth->Instance->PTPTSCR |= ETH_PTPTSCR_TSSTI;
+	while (heth->Instance->PTPTSCR & ETH_PTPTSCR_TSSTI_Msk) {
+		/* spin lock */
+	}
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+
+	irq_unlock(key);
+
+	return 0;
+}
+
+static int ptp_clock_stm32_get(const struct device *dev,
+			      struct net_ptp_time *tm)
+{
+	struct ptp_context *ptp_context = dev->data;
+	struct eth_stm32_hal_dev_data *eth_dev_data = ptp_context->eth_dev_data;
+	ETH_HandleTypeDef *heth = &eth_dev_data->heth;
+	int key;
+
+	key = irq_lock();
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	tm->second = heth->Instance->MACSTSR;
+	tm->nanosecond = heth->Instance->MACSTNR;
+#else
+	tm->second = heth->Instance->PTPTSHR;
+	tm->nanosecond = heth->Instance->PTPTSLR;
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+
+	irq_unlock(key);
+
+	return 0;
+}
+
+static int ptp_clock_stm32_adjust(const struct device *dev, int increment)
+{
+	struct ptp_context *ptp_context = dev->data;
+	struct eth_stm32_hal_dev_data *eth_dev_data = ptp_context->eth_dev_data;
+	ETH_HandleTypeDef *heth = &eth_dev_data->heth;
+	int key, ret;
+
+	if ((increment <= (int32_t)(-NSEC_PER_SEC)) ||
+			(increment >= (int32_t)NSEC_PER_SEC)) {
+		ret = -EINVAL;
+	} else {
+		key = irq_lock();
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		heth->Instance->MACSTSUR = 0;
+		if (increment >= 0) {
+			heth->Instance->MACSTNUR = increment;
+		} else {
+			heth->Instance->MACSTNUR = ETH_MACSTNUR_ADDSUB | (NSEC_PER_SEC + increment);
+		}
+		heth->Instance->MACTSCR |= ETH_MACTSCR_TSUPDT;
+		while (heth->Instance->MACTSCR & ETH_MACTSCR_TSUPDT_Msk) {
+			/* spin lock */
+		}
+#else
+		heth->Instance->PTPTSHUR = 0;
+		if (increment >= 0) {
+			heth->Instance->PTPTSLUR = increment;
+		} else {
+			heth->Instance->PTPTSLUR = ETH_PTPTSLUR_TSUPNS | (-increment);
+		}
+		heth->Instance->PTPTSCR |= ETH_PTPTSCR_TSSTU;
+		while (heth->Instance->PTPTSCR & ETH_PTPTSCR_TSSTU_Msk) {
+			/* spin lock */
+		}
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+
+		ret = 0;
+		irq_unlock(key);
+	}
+
+	return ret;
+}
+
+static int ptp_clock_stm32_rate_adjust(const struct device *dev, float ratio)
+{
+	struct ptp_context *ptp_context = dev->data;
+	struct eth_stm32_hal_dev_data *eth_dev_data = ptp_context->eth_dev_data;
+	ETH_HandleTypeDef *heth = &eth_dev_data->heth;
+	int key, ret;
+	uint32_t addend_val;
+
+	/* No change needed */
+	if (ratio == 1.0) {
```

Comment:
```suggestion
	if (ratio == 1.0f) {
```
Please update everywhere it is applicable

Reply:
Updated, thanks!